compiler/main/checks.cpp

```diff
@@ -261,16 +261,21 @@ bool callSetsSymbol(Symbol* sym, CallExpr* call)
   if (isMoveOrAssign(call)) {
     if (SymExpr* lhs = toSymExpr(call->get(1))) {
       Expr* rhs = call->get(2);
-      // if rhs contains the sym and lhs is a ref
-      if (lhs->isRef() &&
-          !lhs->symbol()->isConstant() &&
-          lhs->symbol()->qualType().getQual() != QUAL_CONST_REF &&
-          exprContainsSymbol(sym, rhs))
-        return true;
-
-      // in some cases, like `localAccess`, the lhs is incorrectly not marked as a ref
-      // so we need to do some more checks
-      if (symExprIsUsedAsRef(lhs, false, checkForMove)) return true;
+      // only check if the symbol is on the rhs
+      if (exprContainsSymbol(sym, rhs)) {
```

Comment:
nit, but putting this as the first one in the chain of `&&`s would have the same effect. No need to change just for my sake though, maybe you wanted it this way to be able to comment that clearly.

Reply:
This was how it was originally, but `exprContainsSymbol` is a precondition for both the `if lhs is a ref` checks and the `lhs is used like a ref but not marked that way` checks